tests/e2e-scality/scripts/modules/uninstall.sh

```diff
@@ -0,0 +1,181 @@
+uninstall_csi_driver() {
+  log "Uninstalling Scality CSI driver..."
+  
+  # Process arguments
+  local DELETE_NS=false
+  local FORCE=false
+  local NAMESPACE="$DEFAULT_NAMESPACE"
+  local IS_CUSTOM_NAMESPACE=false
+  
+  # Parse arguments
+  while [ "$#" -gt 0 ]; do
+    case "$1" in
+      --namespace)
+        NAMESPACE="$2"
+        # Check if a custom namespace was provided (not kube-system)
+        if [ "$NAMESPACE" != "$DEFAULT_NAMESPACE" ]; then
+          IS_CUSTOM_NAMESPACE=true
+        fi
+        shift 2
+        ;;
+      --delete-ns)
+        DELETE_NS=true
+        shift
+        ;;
+      --force)
+        DELETE_NS=true
+        FORCE=true
+        shift
+        ;;
+      *)
+        warn "Unknown parameter for uninstall: $1"
+        shift
+        ;;
+    esac
+  done
+  
+  log "Using namespace: $NAMESPACE"
+  
+  # Check if the namespace exists
+  if ! exec_cmd kubectl get namespace $NAMESPACE &> /dev/null; then
+    warn "Namespace $NAMESPACE does not exist. Nothing to uninstall."
+    return 0
+  fi
+  
+  # Check if the Helm release exists
+  if ! exec_cmd helm status scality-s3-csi -n $NAMESPACE &> /dev/null; then
+    warn "Helm release scality-s3-csi not found in namespace $NAMESPACE."
+  else
+    # Uninstall the Helm release
+    log "Uninstalling Helm release scality-s3-csi from namespace $NAMESPACE..."
+    if ! exec_cmd helm uninstall scality-s3-csi -n $NAMESPACE; then
+      error "Failed to uninstall Helm release from namespace $NAMESPACE. Error code: $ERROR_HELM_UNINSTALL"
+      if [ "$FORCE" = true ]; then
+        warn "Force mode enabled. Continuing with uninstallation despite Helm uninstall failure."
+      else
+        return $ERROR_HELM_UNINSTALL
+      fi
+    else
+      log "Helm release uninstalled successfully from namespace $NAMESPACE."
+    fi
+  fi
+  
+  # Delete AWS credentials secret
+  # Check if secret exists before attempting to delete
+  if exec_cmd kubectl get secret aws-secret -n $NAMESPACE &> /dev/null; then
+    log "Deleting AWS credentials secret from namespace $NAMESPACE..."
+    if ! exec_cmd kubectl delete secret aws-secret -n $NAMESPACE; then
+      error "Failed to delete AWS credentials secret from namespace $NAMESPACE. Error code: $ERROR_SECRET_DELETE"
+      if [ "$FORCE" = true ]; then
+        warn "Force mode enabled. Continuing with uninstallation despite secret deletion failure."
+      else
+        # Non-fatal error, just warn but don't exit
+        warn "Failed to delete AWS credentials secret. You may need to delete it manually."
+      fi
+    else
+      log "AWS credentials secret deleted successfully from namespace $NAMESPACE."
+    fi
+  else
+    warn "AWS credentials secret not found in namespace $NAMESPACE."
+  fi
+  
+  # Only delete namespace if it's a custom namespace (not kube-system)
+  # and the delete flag is set or force mode is enabled
+  if [ "$IS_CUSTOM_NAMESPACE" = true ] && [ "$DELETE_NS" = true -o "$FORCE" = true ]; then
```

Comment:
While initially reading this script I saw the flag `--delete-ns` and thought you were gonna delete `kube-system` 😂